lib/capistrano/tasks/spark.rake

```diff
@@ -0,0 +1,19 @@
+namespace :spark do
+
+  after :deploy, 'spark:update_env'
+
+  # Set the /etc/environment
+  desc 'Update the spark environment variables'
+  task :update_env do
+    on roles(:spark) do
+      # remove any existing entries
+      sudo("sed -i -e '/BEGIN_LD4P_ENV/,/END_LD4P_ENV/{ d; }' /etc/environment")
+      # append new entries
+      sudo("echo '### BEGIN_LD4P_ENV' | sudo tee -a /etc/environment > /dev/null")
```

Comment:
I don't think you need the outer `sudo` in these four lines here.  You **do** need it for `sed`'s inline replace, but there is no reason the `echo` commands would fail without the additional privileges and the right side of the pipes all have their own sudo already.

Reply:
example:
```bash
$ echo "foobar" | sudo tee -a /etc/zubaz
foobar
$ cat /etc/zubaz 
foobar
```